src/array/data_chunk_builder.rs

```diff
@@ -5,12 +5,23 @@ use itertools::Itertools;
 use super::{ArrayBuilderImpl, DataChunk};
 use crate::types::{DataType, DataValue};
 
+/// A helper struct to build a [`DataChunk`].
+///
+/// ## Panic
+///
+/// Panic if there were remaining rows in the builder.
 pub struct DataChunkBuilder {
     array_builders: Vec<ArrayBuilderImpl>,
     size: usize,
     capacity: usize,
 }
 
+impl Drop for DataChunkBuilder {
+    fn drop(&mut self) {
+        assert_eq!(self.size, 0);
```

Comment:
It's better to give error message here.
```suggestion
        assert_eq!(self.size, 0, "dropping non-empty data chunk builder");
```